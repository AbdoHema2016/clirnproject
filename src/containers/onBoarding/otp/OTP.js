import React, {PureComponent} from 'react';
import {
  View,
  Keyboard,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
  Platform,
  TouchableOpacity,
} from 'react-native';
import {connect} from 'react-redux';
import RNOtpVerify from 'react-native-otp-verify';
import CLabel from '../../../components/cLabel';
import CButton from '../../../components/cButton';
import CImage from '../../../components/cImage';
import {Constants} from '../../../utilities';
import {translate} from '../../../Localization';

import Style from './style';
import navigationService from '../../../Navigation/NavigationService';
import HelperFunctions from './helperFunctions';
import {
  sendOtpAction,
  otpFailureAction,
  resendOtpAction,
  resetOtpAction,
} from './redux/actions';
import CNavigationBackButton from '../../../components/cNavigationBackButton';

const {LOCAL_PATH, testIds} = Constants;

class OTP extends PureComponent {
  constructor(props) {
    super(props);
    this.state = this.initialState;
    navigationService.navigation = this.props.navigation;
    HelperFunctions.setProps = this.props;
    HelperFunctions.setInstance = this;
    HelperFunctions.setState = this.state;
  }
  get initialState() {
    return {
      otp: '',
      disableResendButton: true,
      source: this.props.route.params ? this.props.route.params.source : '',
    };
  }
  componentDidMount() {
    if (Platform.OS === 'android') {
      this.focusListener = this.props.navigation.addListener('focus', () => {
        this.startListeningForOtp();
      });
    }
    HelperFunctions.disableResendButton();
    this.navigationOptions();
  }

  startListeningForOtp = () =>
    RNOtpVerify.getOtp()
      .then((p) => RNOtpVerify.addListener(this.otpHandler))
      .catch();

  otpHandler = (message) => {
    if (message) {
      const otp = /(\d{6})/g.exec(message)[1];
      this.setState({otp});
      RNOtpVerify.removeListener();
      Keyboard.dismiss();
    }
  };
  removeListener() {
    RNOtpVerify.removeListener();
  }
  componentWillUnmount() {
    if (Platform.OS === 'android') {
      this.focusListener();
      this.removeListener();
    }
    this.props.resetOtp();
  }
  navigationOptions = () => {
    this.props.navigation.setOptions({
      title: '',
      headerTransparent: true,
      headerLeft: () => (
        <CNavigationBackButton
          title={translate('BACK.BACK_BUTTON_TITLE')}
          backButtonAction={this.goBack}
        />
      ),
    });
  };
  goBack = () => {
    if (this.state.source === 'Edit') {
      this.props.navigation.pop();
      return;
    }
    this.props.navigation.navigate('ContactStack');
  };
  render() {
    const {
      otp: {error, otp, loading},
      access_token,
      token,
    } = this.props;
    const {source} = this.state;
    return (
      <View style={Style.container}>
        <KeyboardAvoidingView
          style={Style.keyboardAvoidingView}
          behavior={Platform.OS === 'ios' ? 'padding' : null}
          keyboardVerticalOffset={Platform.OS === 'android' ? 44 : 0}>
          <TouchableWithoutFeedback
            onPress={Keyboard.dismiss}
            style={Style.keyboardAvoidingView}>
            <View style={Style.dataContainer}>
              <View style={Style.innerContainer}>
                {this.state.source === 'Edit' ? (
                  <View style={Style.infoStageLabel} />
                ) : (
                  <CLabel style={Style.infoStageLabel} text={'3/7'} />
                )}
                <CLabel
                  style={Style.appTitle}
                  text={translate('STRINGS.OTP_HEADING')}
                />
                <View style={Style.buttonsContainer}>
                  {HelperFunctions.otpFeilds(error)}
                </View>

                {otp !== '' && !loading ? (
                  <View style={Style.errorMessageView}>
                    <CImage
                      resizeMode={'contain'}
                      imageStyle={Style.statusIcon}
                      imagePath={
                        error
                          ? LOCAL_PATH.ERROR_ICON
                          : LOCAL_PATH.CHECK_BOX_ICON
                      }
                    />
                    <CLabel
                      style={error ? Style.errorLabel : Style.noErrorLabel}
                      text={
                        error
                          ? translate('STRINGS.OTP_CODE_INCORRECT')
                          : translate('STRINGS.CODE_VERIFIED_SUCCESSFULLY')
                      }
                    />
                  </View>
                ) : null}
              </View>
            </View>
          </TouchableWithoutFeedback>
        </KeyboardAvoidingView>
        {!this.state.disableResendButton && this.state.source !== 'Edit' && (
          <TouchableOpacity
            testID={testIds.resendOtp}
            accessibilityLabel={testIds.resendOtp}
            disabled={this.state.disableResendButton}
            style={Style.didnotReceiveOtpButton}
            onPress={HelperFunctions.resendOtp}>
            <CLabel
              numberOfLines={1}
              style={Style.didnotReceiveOtpText}
              text={
                error
                  ? translate('STRINGS.OTP_SEND_NEW')
                  : translate('STRINGS.OTP_MESSAGE_FAILED')
              }
            />
          </TouchableOpacity>
        )}
        <CButton
          testID={testIds.continue}
          accessibilityLabel={testIds.continue}
          text={translate('LOGIN_SCREEN_STRINGS.CONTINUE_LABEL')}
          textStyle={Style.whiteText}
          onPress={() => HelperFunctions.sendOtp(access_token, token, source)}
          buttonContainerStyle={Style.blackButton}
          loading={loading}
          disabled={HelperFunctions.checkIfAnyOTPFieldIsEmpty()}
          backgroundColor={
            HelperFunctions.checkIfAnyOTPFieldIsEmpty()
              ? Style.blackButton.backgroundColor
              : Style.greenButton.backgroundColor
          }
        />
      </View>
    );
  }
}
const mapStateToProps = ({
  otp,
  personalDetails: {access_token},
  signIn: {token},
}) => ({
  otp,
  access_token,
  token,
});
const mapDispatchToProps = {
  sendOtp: sendOtpAction,
  resendOtp: resendOtpAction,
  emptyOtp: otpFailureAction,
  resetOtp: resetOtpAction,
};
export default connect(mapStateToProps, mapDispatchToProps)(OTP);
