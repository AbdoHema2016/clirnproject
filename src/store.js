import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
