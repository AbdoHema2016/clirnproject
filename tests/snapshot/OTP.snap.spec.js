import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import OTP from '../../src/containers/onBoarding/otp/OTP';

jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-native-otp-verify', () => ({
  getOtp: jest.fn(),
  removeListener: jest.fn(),
}));
jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));
jest.useFakeTimers();
describe('OTP screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
      addListener: jest.fn(),
    },
    addContactetails: jest.fn(),
    resendOtp: jest.fn(),
    emptyOtp: jest.fn(),
    resetOtp: jest.fn(),
    sendOtp: jest.fn(),
    route: {
      params: jest.fn(),
    },
  };
  it('Snapshot OTP', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OTP {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
