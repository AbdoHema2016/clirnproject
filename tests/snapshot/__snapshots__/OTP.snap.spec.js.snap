// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OTP screen Snapshot OTP 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "space-around",
            "width": 724,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8D8C91",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 16,
                "marginTop": 160.07999999999998,
              },
            ]
          }
        >
          3/7
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2B342C",
                "fontFamily": "SofiaPro-Medium",
                "fontSize": 24,
                "fontWeight": "600",
                "width": 324,
              },
            ]
          }
        >
          [missing "en.STRINGS.OTP_HEADING" translation]
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginTop": 68,
              "width": 722,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              testID="OTPInputView"
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": "100%",
                    "justifyContent": "space-between",
                    "width": "100%",
                  }
                }
              >
                <View
                  pointerEvents="none"
                  testID="inputSlotView"
                >
                  <TextInput
                    accessibilityLabel="textInput"
                    accessible={true}
                    allowFontScaling={true}
                    editable={true}
                    keyboardAppearance="default"
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    placeholderTextColor="rgba(226, 226, 226, 1)"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    selectionColor="#02BF8E"
                    style={
                      Array [
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 4,
                            "color": "#2B342C",
                            "fontFamily": "SofiaProRegular",
                            "fontSize": 24,
                            "height": 48,
                            "marginLeft": 8,
                            "textAlign": "center",
                            "width": 48,
                          },
                        ],
                      ]
                    }
                    testID="textInput"
                    textContentType="none"
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
                <View
                  pointerEvents="none"
                  testID="inputSlotView"
                >
                  <TextInput
                    accessibilityLabel="textInput"
                    accessible={true}
                    allowFontScaling={true}
                    editable={true}
                    keyboardAppearance="default"
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    placeholderTextColor="rgba(226, 226, 226, 1)"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    selectionColor="#02BF8E"
                    style={
                      Array [
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 4,
                            "color": "#2B342C",
                            "fontFamily": "SofiaProRegular",
                            "fontSize": 24,
                            "height": 48,
                            "marginLeft": 8,
                            "textAlign": "center",
                            "width": 48,
                          },
                        ],
                      ]
                    }
                    testID="textInput"
                    textContentType="none"
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
                <View
                  pointerEvents="none"
                  testID="inputSlotView"
                >
                  <TextInput
                    accessibilityLabel="textInput"
                    accessible={true}
                    allowFontScaling={true}
                    editable={true}
                    keyboardAppearance="default"
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    placeholderTextColor="rgba(226, 226, 226, 1)"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    selectionColor="#02BF8E"
                    style={
                      Array [
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 4,
                            "color": "#2B342C",
                            "fontFamily": "SofiaProRegular",
                            "fontSize": 24,
                            "height": 48,
                            "marginLeft": 8,
                            "textAlign": "center",
                            "width": 48,
                          },
                        ],
                      ]
                    }
                    testID="textInput"
                    textContentType="none"
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
                <View
                  pointerEvents="none"
                  testID="inputSlotView"
                >
                  <TextInput
                    accessibilityLabel="textInput"
                    accessible={true}
                    allowFontScaling={true}
                    editable={true}
                    keyboardAppearance="default"
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    placeholderTextColor="rgba(226, 226, 226, 1)"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    selectionColor="#02BF8E"
                    style={
                      Array [
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 4,
                            "color": "#2B342C",
                            "fontFamily": "SofiaProRegular",
                            "fontSize": 24,
                            "height": 48,
                            "marginLeft": 8,
                            "textAlign": "center",
                            "width": 48,
                          },
                        ],
                      ]
                    }
                    testID="textInput"
                    textContentType="none"
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
                <View
                  pointerEvents="none"
                  testID="inputSlotView"
                >
                  <TextInput
                    accessibilityLabel="textInput"
                    accessible={true}
                    allowFontScaling={true}
                    editable={true}
                    keyboardAppearance="default"
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    placeholderTextColor="rgba(226, 226, 226, 1)"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    selectionColor="#02BF8E"
                    style={
                      Array [
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 4,
                            "color": "#2B342C",
                            "fontFamily": "SofiaProRegular",
                            "fontSize": 24,
                            "height": 48,
                            "marginLeft": 8,
                            "textAlign": "center",
                            "width": 48,
                          },
                        ],
                      ]
                    }
                    testID="textInput"
                    textContentType="none"
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
                <View
                  pointerEvents="none"
                  testID="inputSlotView"
                >
                  <TextInput
                    accessibilityLabel="textInput"
                    accessible={true}
                    allowFontScaling={true}
                    editable={true}
                    keyboardAppearance="default"
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    onKeyPress={[Function]}
                    placeholder=""
                    placeholderTextColor="rgba(226, 226, 226, 1)"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    selectionColor="#02BF8E"
                    style={
                      Array [
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderRadius": 4,
                            "color": "#2B342C",
                            "fontFamily": "SofiaProRegular",
                            "fontSize": 24,
                            "height": 48,
                            "marginLeft": 8,
                            "textAlign": "center",
                            "width": 48,
                          },
                        ],
                      ]
                    }
                    testID="textInput"
                    textContentType="none"
                    underlineColorAndroid="rgba(0,0,0,0)"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "height": 52,
          "justifyContent": "center",
          "shadowColor": "#DDDDDD",
          "shadowOffset": Object {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 4,
        },
        Object {
          "backgroundColor": "#565655",
          "bottom": 158.74599999999998,
          "marginTop": 58,
          "position": "absolute",
        },
      ]
    }
  >
    <View
      accessibilityLabel="Continue"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#565655",
          "borderRadius": 4,
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": 718,
        }
      }
      testID="Continue"
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "SofiaProRegular",
              "fontSize": 17,
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        [missing "en.LOGIN_SCREEN_STRINGS.CONTINUE_LABEL" translation]
      </Text>
    </View>
  </View>
</View>
`;
